vertical-pod-autoscaler/templates/recommender-pod-monitor.yaml

```diff
@@ -0,0 +1,20 @@
+{{- if .Values.recommender.servicemonitor.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+metadata:
+  name: {{ include "vpa.fullname" . }}-recommender
+  labels:
+    app.kubernetes.io/component: recommender
+    {{- include "vpa.labels" . | nindent 4 }}
+spec:
+  podMetricsEndpoints:
+  - interval: {{ .Values.recommender.servicemonitor.scrapeInterval }}
+    port: metrics
+  namespaceSelector:
+    matchNames:
+    - "{{ .Release.Namespace }}"
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: recommender
```

Comment:
I think you should also use the selector labels here. `app.kubernetes.io/component: recommender` is quite broad imo

Reply:
actually, it's `vpa.selectorLabels` you need, otherwise it won't match anything afaik